src/storage/filesystem-adapter.test.ts

```
@@ -0,0 +1,150 @@
+      expect(defaultAdapter['filePath']).toBe(path.join(process.cwd(), 'run-cache-data.json'));
+    });
+
+    it('should use custom file path if provided', () => {
+      const customPath = path.join(process.cwd(), 'custom-cache.json');
+      const customAdapter = new FilesystemAdapter({ filePath: customPath });
+      expect(customAdapter['filePath']).toBe(customPath);
+    });
+
+    it('should throw error if not in Node.js environment', () => {
+      // Temporarily remove process.versions.node
+      const versions = process.versions;
```

ESLint

[error] Use object destructuring.

(prefer-destructuring)